algorithms/branching.py

```diff
@@ -82,16 +84,19 @@ def __branching(G: 'Graph', colors: 'Dict[Int, List[Vertex]]', C: 'Int', D: 'Lis
     num_isomorphisms = 0
     for y in g1:
         # Make a copy of everything before creating a new branch
-        G_backup, max_colornum_backup = G.backup()
+        G_backup, max_colornum_backup, colors_backup = G.backup()
         G.max_colornum += 1
         D_copy = D.copy()
         I_copy = I.copy()
         y.colornum = G.max_colornum
+        # Update colors of graph
+        G.colors[C].remove(y)
+        G.colors.setdefault(G.max_colornum, list()).append(y)
         I_copy.append(y)
         num_isomorphisms += count_isomorphisms(G, D_copy, I_copy, count_flag, color_refinement_method)
         if not count_flag and num_isomorphisms > 0:
             return True
-        G.revert(G_backup, max_colornum_backup)
+        G.revert(G_backup, max_colornum_backup, colors_backup)
```

Comment:
G_color_list_backup i.p.v. G_backup maakt het voor mij duidelijker wat er in de variabele staat.

Reply:
Veranderd.